eclair-core/src/main/scala/fr/acinq/eclair/db/Databases.scala

```diff
@@ -162,6 +174,71 @@ object Databases extends Logging {
         }
       }
 
+      safetyChecks_opt foreach { initChecks =>
+
+        PgUtils.inTransaction { connection =>
+          using(connection.createStatement()) { statement =>
+
+            def checkMaxAge(name: String, maxAge: FiniteDuration, sqlQuery: String): Unit = {
+              import ExtendedResultSet._
+              val smallestAge_opt = statement
+                .executeQuery(sqlQuery)
+                .headOption // sql max() will always return a result, with perhaps a null value if there was no records
+                .flatMap(_.getTimestampNullable("max"))
+                .map(ts => TimestampMilli.now() - TimestampMilli.fromSqlTimestamp(ts))
+              require(smallestAge_opt.isDefined, s"db check failed: no $name found")
+              require(smallestAge_opt.get <= maxAge, s"db check failed: most recent $name is too old (${smallestAge_opt.get.toMinutes} minutes > ${maxAge.toMinutes} minutes)")
```

Comment:
nit: I thought we chose to use `sys.exit` for that kind of scenarios (it's easier to search for them in the codebase than filter all `require`). Same for `checkMinCount`.

Also, in my understanding `require` should be used mostly to check for logic bugs, incorrect inputs or invalid assumptions (so the check that `smallestAge_opt.isDefined` does make sense as a `require`).

But it would make it much harder to unit test...so probably better to ignore this ¯\_(ツ)_/¯

Reply:
We're not very consistent in that file. I could `throw` instead.

Reply:
Let's keep it like that, it's not very important.